common/utils.py

```diff
@@ -0,0 +1,53 @@
+from datetime import date, timedelta
+
+
+def _get_day_arg(day_arg):
+    """
+    This function takes in a day argument and returns the right timedelta
+    days argument for it.
+    For example, yesterday will be returned as -1, so that get_date function
+    can convert it to the appropriate pythonic date object.
+    We also want to be able to convert weekdays and some date format here.
+    """
+    day_arg = day_arg.strip().lower()
+
+    day_to_num_dict = {
+        'yesterday': -1,
+        'today': 0,
+        'tomorrow': 1
+    }
+
+    day_num = day_to_num_dict.get(day_arg)
+    # More logic if a date object or weekday value is passed in.
+    if day_num is not None:
+        return day_num
+    else:
+        days = ['monday', 'tuesday', 'wednesday', 'thursday', 'friday',
+                'saturday', 'sunday']
+
+        if day_arg in days:
+            # In python, Monday is 0 and Sunday is 6
+            current_weekday_num = date.today().weekday()
+            day_index = days.index(day_arg)
+
+            if current_weekday_num == day_index:
+                # The weekday passed in is today
+                return day_to_num_dict['today']
+            elif current_weekday_num > day_index:
+                # It is a previous weekday
+                return 0 - (current_weekday_num - day_index)
```

Comment:
if you use `day_index - current_weekday_num` here, then you wouldn't have to use the `0 -` part